### ISSUE TITLE
RFE: podman generate kube storage usability

### ISSUE THREAD

**Issue by langdon** [containers/podman#15000]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

**Steps to reproduce the issue:**

follow: https://docs.ckan.org/en/2.9/maintaining/installing/install-from-docker-compose.html using podman-compose then generate the kube yaml

**Output of `podman version`:**

```
$ podman-compose --version
['podman', '--version', '']
using podman version: 4.1.1
podman-composer version  1.0.3
podman --version 
podman version 4.1.1
exit code: 0
```

**Comment by rhatdan** [comment]
@langdon Could you submit an example kube.yaml where you would want the warning?

**Comment by langdon** [comment]
Yeah:

* input -> [`docker-compose.yml`](https://github.com/ckan/ckan/blob/master/contrib/docker/docker-compose.yml)
* what podman generated; mostly, includes edits to work with the volume manifest -> [`ckan-manifest.yml`](https://github.com/BU-Spark/infra-public-data-portal/blob/main/contrib/kubernetes/ckan-manifest.yml)
* what I needed to create -> [`ckan-volume-manifest.yml`](https://github.com/BU-Spark/infra-public-data-portal/blob/main/contrib/kubernetes/ckan-volume-manifest.yml)

basically, when podman makes `ckan-manifest.yml` a warning that the user still needs to create something like `ckan-volume-manifest.yml`. Please keep in mind that the examples ^^ may not work correctly that are still WIP.

**Comment by rhatdan** [comment]
@umohnani8 PTAL

**Comment by umohnani8** [comment]
@langdon we generate PVCs for volumes you have used when creating containers/pods with podman. However this is not done automatically when you run `podman generate kube` on a pod or container - to generate the PVC yamls you need to pass in the volume names as well `podman generate kube volName` and something like this will be generated for you

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    volume.podman.io/driver: local
  creationTimestamp: "2022-08-02T04:05:53Z"
  name: pvc-test
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
status: {}
```

Is this what you were looking for or is there something else you would like podman to support or warn about?

The `podman volume ls` command will show you all the volumes you currently have created with podman. Generate kube will also create a multi yaml file with the PVC and pod definition when you pass both to the generate kube command, e.g `podman generate kube volName ctrName`.

**Comment by langdon** [comment]
Sorry, been very busy. Will review by EOW

**Comment by umohnani8** [comment]
@langdon please reopen this if the suggestion in https://github.com/containers/podman/issues/15000#issuecomment-1226743111 isn't what you were looking for.
